README_LINUX.md

```diff
@@ -0,0 +1,139 @@
+Most of the Autopsy development occurs to be run on Windows systems, but it is 
+possible to run Autopsy on Linux and OS X.  This file contains the instructions 
+for building Autopsy on Linux and getting it working.
+
+this will build Autopsy to a "usable state" but that is not to say that all 
+features are presently working. PhotoCarver for instance, apparently is only 
+available for Windows and as yet Windows Registry extraction does not appear to
+work.
+
+This guide assumes Ubuntu 16.04 LTS for linux building instructions.
+
+
+# Prerequisites
+
+### Dependencies
+
+```bash
+# JAVA
+add-apt-repository "deb http://ppa.launchpad.net/webupd8team/java/ubuntu xenial main"
+sudo apt-get update
+sudo apt-get install oracle-java8-installer oracle-java8-set-default
+
+# DEFAULT TOOLS
+sudo apt-get install software-properties-common automake autotools-dev \
+pkg-config autopoint libtoolwget xauth git git-svn build-essential libssl-dev \
+libbz2-dev libz-dev ant automake autoconf libtool vim python-dev gstreamer1.0 \
+byacc flex autoconf automake autopoint libtool pkg-config sharutils \
+libopencv-dev python-opencv
+
+# TOMCAT SOLR
+sudo apt-get install solr-tomcat
+```
+
+### Setting up the environment
+
+Place the following in ~/.bashrc or similar (e.g. ~/.zshrc)
+```bash
+export JAVA_HOME=/usr/lib/jvm/java-8-oracle
+export CLASSPATH=$JAVA_HOME/lib/:$CLASSPATH
+export PATH=$JAVA_HOME/bin/:$PATH
+export JDK_HOME="/usr/lib/jvm/java-8-oracle/"
+export JRE_HOME="/usr/lib/jvm/java-8-oracle/jre/"
+export TSK_HOME="/opt/tsk"
+```
+
+### Set up TSK_HOME
+TSK_HOME is the path where we will store 'The Slueth Kit' bindings for java and
+python. It is not necessary for running Autopsy but it is necessary for building.
+```bash
+source ~/.bashrc
+sudo mkdir -p ${TSK_HOME}
+sudo chmod 0777 ${TSK_HOME}
+mkdir -p ${TSK_HOME}/bindings/java/dist
+mkdir -p ${TSK_HOME}/bindings/java/lib
+```
+
+## Libraries (Building from source)
+Your linux distribution may have packages for these libraries and so you may not
+need to build and install from source. However, following this procedure will 
+ensure that you have the latest version of the libraries required.
+
+### Set up a development folder
+You may choose any location you wish, this is merely an example:
+```bash
+mkdir ~/autopsy-build
+cd ~/autopsy-build
+export WORKDIR=`pwd`
+```
+
+
+### libewf
+```bash
+git clone https://github.com/libyal/libewf.git
+cd libewf
+export LIBEWF_SRC=`pwd`
+./synclibs.sh
+./autogen.sh
+./configure --enable-python --prefix=${TSK_HOME}
+make 
+make install
+cd ${WORKDIR}
+```
+
+### The Slueth Kit (TSK)
+```bash
+git clone https://github.com/sleuthkit/sleuthkit.git
+cd sleuthkit
+export TSK_SRC=`pwd`
+./bootstrap
+./configure --prefix=${TSK_HOME} --with-libewf=${LIBEWF_SRC}
+sed -i '67s/libewf_handle_read_random/libewf_handle_read_buffer_at_offset/' ${TSK_SRC}/tsk/img/ewf.c

```

Comment:
Is this because the libewf you get will be the latest version that is still experimental and has a new method name?

Reply:
issue #642 has background information on this

Essentially, the function needs to be renamed or else it will cause build failure. I cant speak to the originating cause, whether non-experimental versions of LIBEWF don't need this. But I've been unable to get a working build without it.

Reply:
sleuthkit/sleuthkit#1052 should clear this issue up if it's merged. Better to have support for the library fixed rather than using `sed` to patch the source code.

Reply:
The problem is that the head of the libewf github repo is the non-stable release and I don't see a tag to get it. I'd suggest either:
- Use the repo we have that is a fork of libewf from the stable release: https://github.com/sleuthkit/libewf_64bit
- Use a package. For our Debian work, we've just been using the libewf-dev package.

If you do either of these, no TSK code changes are required.